contracts/helper/RebalancingSetCTokenBidder.sol

```diff
@@ -0,0 +1,360 @@
+
+        // Ensure allowances and transfer auction tokens or underlying from user
+        depositComponents(
+            combinedTokenArray,
+            inflowArray
+        );
+
+        // Bid in auction
+        rebalanceAuctionModule.bidAndWithdraw(
+            address(_rebalancingSetToken),
+            _quantity,
+            _allowPartialFill
+        );
+
+        // Withdraw auction tokens or underlying to user
+        withdrawComponentsToSender(
+            combinedTokenArray
+        );
+
+        // Log bid placed with Eth event
+        emit BidPlacedCToken(
+            address(_rebalancingSetToken),
+            msg.sender
+        );
+    }
+
+    /*
+     * Get token inflows and outflows and combined token array denominated in underlying required
+     * for bid for a given rebalancing Set token.
+     *
+     * @param _quantity               The amount of currentSet to be rebalanced
+     * @param _rebalancingSetToken    The rebalancing Set Token instance
+     * @return inflowArray            Array of amount of tokens inserted into system in bid
+     * @return outflowArray           Array of amount of tokens returned from system in bid
+     * @return combinedTokenArray     Array of token addresses
+     */
+    function getBidPriceAndAddressArray(
+        uint256 _quantity,
+        IRebalancingSetToken _rebalancingSetToken
+    )
+        public
+        view
+        returns (uint256[] memory, uint256[] memory, address[] memory)
+    {
+        // Get token addresses
+        address[] memory combinedTokenArray = _rebalancingSetToken.getCombinedTokenArray();
+
+        // Get inflow and outflow arrays for the given bid quantity
+        uint256[] memory inflowArray;
+        uint256[] memory outflowArray;
+        (
+            inflowArray,
+            outflowArray
+        ) = _rebalancingSetToken.getBidPrice(_quantity);
+
+        // Loop through the combined token addresses array and replace with underlying address
+        for (uint256 i = 0; i < combinedTokenArray.length; i++) {
+            address currentComponent = combinedTokenArray[i];
+
+            if (currentComponent == address(targetCToken)) {
+                combinedTokenArray[i] = underlyingAddress;
+
+                // Replace inflow and outflow with required amount of underlying. 
+                // Calculated as cToken quantity * exchangeRate / scalingFactor.
+                uint256 exchangeRate = targetCToken.exchangeRateStored();
+                uint256 currentInflowQuantity = inflowArray[i];
+                uint256 currentOutflowQuantity = outflowArray[i];
+
+                inflowArray[i] = currentInflowQuantity.mul(exchangeRate).div(scalingFactor);
+                outflowArray[i] = currentOutflowQuantity.mul(exchangeRate).div(scalingFactor);
+
+                // Check for rounding error and add 1 if needed
+                inflowArray[i] = inflowArray[i].mul(scalingFactor).div(exchangeRate) >= currentInflowQuantity
```

Comment:
In what cases would there be a rounding error? 

There may be a better way to handle this.

Reply:
Seems like when converting from cToken amount to underlying. Since required underlying amount is less than cToken amount, and if its an odd number

I copied the logic from instadapp https://github.com/InstaDApp/smart-contract/blob/master/contracts/bridges/MakerCompound.sol#L394